_blog/2024-02-09-using-clerk-in-rust.mdx

```diff
@@ -0,0 +1,1014 @@
+
+Here is the [source code](https://github.com/sourabpramanik/issue-tracker) of the complete project if you want to follow along.
+
+To start this project I have used the template from Shuttle’s [example repository](https://github.com/shuttle-hq/shuttle-examples) that has a barebone setup of Vite, Shuttle, Actix web, and Clerk. 
+
+The back-end uses [clerk-rs](https://github.com/DarrenBaldwin07/clerk-rs) crate to create a connection between the back-end application and Clerk project. This crate has many different APIs for different use cases, in the template there is back-end API to fetch all users record using `clerk-rs` user API. We can easily verify user's identity trying to access protected routes before hand using the `ClerkMiddleware` from the crate.
+
+I have additionally used Postgres for the database which will run in a docker container so make sure you also have docker [installed](https://docs.docker.com/engine/install/).
+
+
+# Let's get started
+
+First, we need to clone the template from the Shuttle’s example repository using Shuttle CLI:
+
+```bash
+cargo shuttle init --from shuttle-hq/shuttle-examples --subfolder actix-web/clerk
+
+```
+
+You will get a few prompts to create a new project using this template. Make sure you provide a unique name because when you deploy this project on Shuttle this project name will become the sub-domain. And once a name is taken it cannot be used for other project as well:
+
+```bash
+What do you want to name your project?
+It will be hosted at ${project_name}.shuttleapp.rs, so choose something unique!
+✔ Project name · <ENTER_PROJECT_NAME>
+Where should we create this project?
+✔ Directory · <ENTER_THE_DIRECTORY_NAME>
+✔ Claim the project name <PROJECT_NAME> by starting a project container on Shuttle? <Y/N>
+
+```
+
+After that, `cd` into the project directory:
+
+```bash
+cd {project_name}
+
+```
+
+## Setup a new application in Clerk
+
+Head on to Clerk’s official website, sign in or sign up if don’t have an account, and create a new application in the dashboard.
+
+Give a name to your project and select the providers using which users can sign-in or sign-up in your application.
+
+After that, you will get a **Publishable Key** and a **Secret key** which we will use later in our application.
+
+[https://gist.github.com/assets/61370770/7fb52565-605c-42e9-9a6a-3fc7e4855696](https://gist.github.com/assets/61370770/7fb52565-605c-42e9-9a6a-3fc7e4855696)
+
+## Build the back-end
+
+Now we can start updating the existing back-end code to write and build our own endpoints for our application.
+
+`cd` into the backend:
+
+```bash
+cd backend
+
+```
+
+Copy the contents of `Secrets.example.toml` into a new `Secrets.toml` file:
+
+```bash
+touch Secrets.toml
+cp Secrets.example.toml Secrets.toml
+
+```
```

Comment:
The cd command can be combined with the code block below to make it more compact